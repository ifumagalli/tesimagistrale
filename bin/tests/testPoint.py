from dolfin import *

N = 5
# due spazi diversi in tutto (ma entrambi CG)
V = VectorFunctionSpace(UnitSquare(N,N),'CG',2)
Q = FunctionSpace(UnitInterval(N),'CG',1)

v = Function(V)
q = Function(Q)

v_expr = Expression(('sin(2*pi*x[0])','5*exp(x[1])+x[0]'))
q_expr = Expression('sin(2*pi*x[0])')

v = project(v_expr,V)
q = project(q_expr,Q)

pv = Point(0.4,0.2)
pq = Point(0.4)
print("v_expr(0.5,0.2) = {0},   q_expr(0.5) = {1}".format(v_expr(pv), q_expr(pq)))

val_q = q(pq)
val_v = v(pv)

print("v(0.5,0.2) = {0},   q(0.5) = {1}".format(val_v, val_q))

print("Test superato!")
