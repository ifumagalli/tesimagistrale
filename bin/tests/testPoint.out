*** -------------------------------------------------------------------------
*** Warning: UnitSquare has been deprecated in DOLFIN version 1.1.0.
*** The class UnitSquare has been replaced by UnitSquareMesh.
*** -------------------------------------------------------------------------

*** -------------------------------------------------------------------------
*** Warning: UnitInterval has been deprecated in DOLFIN version 1.1.0.
*** The class UnitInterval has been replaced by UnitIntervalMesh.
*** -------------------------------------------------------------------------

Solving linear system of size 242 x 242 (PETSc Krylov solver).
Solving linear system of size 6 x 6 (PETSc Krylov solver).
v_expr(0.5,0.2) = [ 0.58778525  6.50701379],   q_expr(0.5) = 0.587785252292
Traceback (most recent call last):
  File "testPoint.py", line 21, in <module>
    val_q = q(pq)
  File "/opt/libs/dolfin/1.2.0/gcc/4.8.0/lib/python2.7/site-packages/dolfin/functions/function.py", line 372, in __call__
    raise TypeError("expected scalar arguments for the coordinates")
TypeError: expected scalar arguments for the coordinates
